lib/modules/datasource/npm/get.ts

```diff
@@ -56,19 +63,54 @@ export async function getDependency(
 
   // Now check the persistent cache
   const cacheNamespace = 'datasource-npm';
-  const cachedResult = await packageCache.get<NpmDependency>(
+  const cachedResult = await packageCache.get<CachedNpmDependency>(
     cacheNamespace,
     packageUrl
   );
-  // istanbul ignore if
   if (cachedResult) {
-    return cachedResult;
+    if (cachedResult.cacheData) {
+      const softExpireAt = DateTime.fromISO(
+        cachedResult.cacheData.softExpireAt
+      );
+      if (softExpireAt > DateTime.local()) {
```

Comment:
```suggestion
      if (softExpireAt.isValid && softExpireAt > DateTime.local()) {
```
`DateTime.fromISO(undefined) > DateTime.local() === false`
sanity check